File: Tags/AutoAssignInterface.vb
Imports Microsoft.Office.Interop.Outlook
Imports UtilitiesVB

Public Interface IAutoAssign

    Function AutoFind(objItem As Object) As IList(Of String)

    Function AddChoicesToDict(olMail As Microsoft.Office.Interop.Outlook.MailItem,
                              prefixes As List(Of IPrefix),
                              prefixKey As String) As IList(Of String)

    Function AddColorCategory(prefix As IPrefix, categoryName As String) As Category

    ReadOnly Property FilterList As List(Of String)

End Interface

File: Tags/AutoAssignInterface.cs
using System.Collections.Generic;
using Microsoft.Office.Interop.Outlook;
using UtilitiesVB;

namespace Tags
{

    public interface IAutoAssign
    {

        IList<string> AutoFind(object objItem);

        IList<string> AddChoicesToDict(MailItem olMail, List<IPrefix> prefixes, string prefixKey, string currentUserEmail);

        Category AddColorCategory(IPrefix prefix, string categoryName);

        List<string> FilterList { get; }

    }
}